gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -1048,9 +1049,6 @@ private void submitJob(DagNode<JobExecutionPlan> dagNode) {
     /**
      * Method that defines the actions to be performed when a job finishes either successfully or with failure.
      * This method updates the state of the dag and performs clean up actions as necessary.
-     * TODO : Dag should have a status field, like JobExecutionPlan has. This method should update that field,
```

Comment:
Seems we remove failedDagIdsFinishRunning, failedDagIdsFinishAllPossible in this PR as well. But I don't have enough context for what these two are used for right now and how current implementation can support the same use case. 
Should we separate this into a separate PR? Or at lease, please describe the change in PR description.

Reply:
So those behaviors still stay, but i do the processing during cleanup since each dag stores a reference to their status so those sets are no longer needed.